fv3core/utils/mpi.py

```diff
@@ -2,3 +2,66 @@
     from mpi4py import MPI
 except ImportError:
     MPI = None
+from typing import Any
+
+
+def is_parallel_context() -> bool:
+    """Checks if we are in a parallel context, meaing the size of the
+    world communicatior is greater than 1.
+
+    Returns:
+        bool: if the context is parallel
+    """
+    return bool(MPI is not None and MPI.COMM_WORLD.Get_size() > 1)
+
+
+def get_size() -> int:
+    """Get the size of the MPI COMM_WORLD communicator if it is defined.
+
+    Returns:
+        int: size
+    """
+    return MPI.COMM_WORLD.Get_size() if MPI else 0
+
+
+def get_rank() -> int:
+    """Get the current rank in the MPI COMM_WORLD communicator if it is defined.
+
+    Returns:
+        int: rank
+    """
+    return MPI.COMM_WORLD.Get_rank() if is_parallel_context() else -1
+
```

Comment:
We could consider using some  of the functionality of the submodule fvgfs-util, and accessing through the 'comm' object, but it may not be available in the scope we need it currently (I think that was some of the idea Johann was proposing). So that might be a subsequent step, but I am ok with putting it here for now.